### ISSUE TITLE
Multiple Label classification with Label Encoded.

### ISSUE THREAD

**Issue by liuauto** [marcotcr/lime#163]
Hi, 

Recently I want to try LIME on a multiple label classification model built with keras. 

The problem for me is that when I built the model using keras, the labels must be encoded into one-hot vectors. 
``
  encoder = LabelBinarizer()
``
``
  train_label = encoder.fit_transform(train_df['LABEL'].values)
``
``
  test_label = encoder.transform(test_df['LABEL'].values)
``
Now train_label and test_label both become arrays of one hot encoded vectors. 

But LIME only accepts scalar valued labels. So it is not possible to use class_names as input. I tried the Keras API KerasClassifier which allows me to create a sklearn like interface for the model predict() but still, there is no way to convert the vectorized label. 

I checked the tutorial provided in the repo (https://github.com/marcotcr/lime/blob/master/doc/notebooks/Lime%20-%20multiclass.ipynb) but it was built on top of sklearn datasets, which handles the mapping from the string labels to one-hot labels. 

Is there any way to do the conversion in order to use LIME? 

Thanks,
Jie

**Comment by marcotcr** [comment]
LIME doesn't care how you encode your labels, but you must provide a predict_proba function that maps from  inputs to arrays of length L, where L is the number of labels.
If your predict function just returns the predicted label with no probabilities, you need to map that output with something like the following, and use it with LIME.
```
def new_predict_proba(x):
  labels = model.predict(x)
  n_labels = np.zeros((x.shape[0], L), dtype='int')
  for i, l in enumerate(labels):
     n_labels[i][l] = 1
  return n_labels
```

**Comment by liuauto** [comment]
Thank you for the information! It turns out that I originally supplied an element of pandas Series as an input to explain_instance. Use test_text.values[idx] and then everything works. 

Thanks,
Jie
